x-pack/platform/plugins/shared/inference/common/utils/get_request_aborted_signal.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Observable } from 'rxjs';
+
+export function getRequestAbortedSignal(aborted$: Observable<void>): AbortSignal {
```

Comment:
I find the naming a bit confusing - it’s not just returning an abort signal, but also initiating a subscription, which means you’ll need to unsubscribe at some point as well.

Reply:
as above I think this is fine because aborted$ completes when the request ends , but I'll make the function signature more precise to prevent people from subscribing to random observables.